src/contrib/coh_gamma/COHNCGamma_ext/NCgamma_Hadronic_Current_R.cpp

```diff
@@ -0,0 +1,862 @@
+}
+
+
+
+
+
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////////
+
+
+Hadronic_Current_R::Hadronic_Current_R() {
+    this->c_i = NCgamma_Param::c_i;
+}
+
+void Hadronic_Current_R::set_tr_dir(Array4x4 &tr) {}
+
+void Hadronic_Current_R::set_tr_crs(Array4x4 &tr) {}
+
+void Hadronic_Current_R::setQ(const std::vector<double> &q_in) {
+    Hadronic_Current_R::q = q_in;
+    double q2 = q_in[0] * q_in[0] - q_in[1] * q_in[1] -q_in[2] * q_in[2] -q_in[3] * q_in[3];
+    this->setFF(q2);
+}
+
+void Hadronic_Current_R::setKg(const std::vector<double> &kg_in) {
+    Hadronic_Current_R::kg = kg_in;
+}
+
+void Hadronic_Current_R::setP(std::vector<double> p_in) {
+    Hadronic_Current_R::p = p_in;
+    this->p0 = p_in[0];
+    double pp_aux[] = {p_in[0], -p_in[1], -p_in[2], -p_in[3]};
+    this->pp.assign(pp_aux, pp_aux + sizeof(pp_aux) / sizeof(double) );
+}
+
+std::complex<double> Hadronic_Current_R::getR(int i, int j) {
+    return 0;
+}
+
+void Hadronic_Current_R::setFF(double q2) {}
+
+Hadronic_Current_R::~Hadronic_Current_R() {}
+
+
+
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////////
+
+
+Hadronic_Current_R_Sum::Hadronic_Current_R_Sum(const std::vector<Hadronic_Current_R *> &vectorOfCurrents)
+        : vector_of_currents(vectorOfCurrents) {}
+
+void Hadronic_Current_R_Sum::setQ(const std::vector<double> &q_in) {
+    for (int i = 0; i < static_cast<unsigned int>(this->vector_of_currents.size()); ++i) {
+        vector_of_currents[i]->setQ(q_in);
+    }
+}
+
+void Hadronic_Current_R_Sum::setKg(const std::vector<double> &kg_in) {
+    for (int i = 0; i < static_cast<unsigned int>(this->vector_of_currents.size()); ++i) {
+        vector_of_currents[i]->setKg(kg_in);
+    }
+}
+
+void Hadronic_Current_R_Sum::setP(std::vector<double> p_in) {
+    for (int i = 0; i < static_cast<unsigned int>(this->vector_of_currents.size()); ++i) {
+        vector_of_currents[i]->setP(p_in);
+    }
+}
+
+std::complex<double> Hadronic_Current_R_Sum::getR(int i, int j) {
+    std::complex<double> R  (0.0,0.0);
+    for (int k = 0; k < static_cast<unsigned int>(this->vector_of_currents.size()); ++k) {
+        R += vector_of_currents[k]->getR(i,j);
+    }
+    return R;
+}
+
+void Hadronic_Current_R_Sum::setFF(double p_in) {
+    for (int i = 0; i < static_cast<unsigned int>(this->vector_of_currents.size()); ++i) {
+        vector_of_currents[i]->setFF(p_in);
+    }
+}
+
+
+
+
+
+//////////////////////////////////////////////////////////////////////////////////////////
+//////////////////////////////////////////////////////////////////////////////////////////
+
+
+
+Hadronic_Current_R_Delta::Hadronic_Current_R_Delta(Nuclear_FF *nuclearFf) : nuclearFF(nuclearFf) {
+    this->mn = NCgamma_Param::mp;
```

Comment:
Again with hard coded values that might not match values used elsewhere.  Consistency with PDGLibrary values.